wordpress-mobile/WordPress-Android
# 🤖 Add backend-driven Dynamic Cards filtering

Fixes #19870

**Depends on: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2935**

## Description
This PR calls the updates dashboard cards endpoint passing the extra parameters (device_id, platform, build_number, marketing_version, identifier) and removes the local dynamic cards filtering logic that was based on the deleted `remote_feature_flag` response field.

-----

## To Test:
- Sandbox your emulator with D133371-code (instructions paqN3M-5m-p2)
- Apply the patch [version239.patch](https://github.com/wordpress-mobile/WordPress-Android/files/13818331/version239.patch)
- **Verify** that the domain management card does not appear
- Apply the patch [version240.patch](https://github.com/wordpress-mobile/WordPress-Android/files/13818336/version240.patch)
- **Verify** that the domain management card appears

## Note 
Please remove the `Do Not Merge` label and merge after:
* the backend change is deployed D133371-code
* FluxC PR is merged https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2935 into `trunk`
* The `wordPressFluxCVersion` reference on this PR is updated to point to the latest FluxC `trunk`

-----

## Regression Notes

1. Potential unintended areas of impact

    - My Site Dashboard Screen

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

    - Updated existing tests

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
